pkg/planner/core/logical_plan_builder.go

```diff
@@ -4594,6 +4594,7 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 			}
 		}
 	}
+	countCnt := len(columns) + 1 // +1 for an extra handle column
```

Comment:
It seems that we need an `intest.Assert` for the if block at line#4643 to make sure that clustered index should not be added as the extra one.